unison-core/src/Unison/Var.hs

```diff
@@ -137,7 +137,7 @@ reset :: Var v => v -> v
 reset v = typed (typeOf v)
 
 unqualifiedName :: Var v => v -> Text
-unqualifiedName = last . Name.segments' . name
+unqualifiedName = fromMaybe "" . lastMay . Name.segments' . name
```

Comment:
If the incoming `v` has an empty name, this isn't the place to handle that.

Reply:
Maybe crashing loudly would be better?

Reply:
Yeah, I'd replace that `fromMaybe ""` with `fromMaybe (error "unqualifiedName of empty name")`. That will preserve the old behavior rather than returning what the caller might consider a bogus value and leading to issues elsewhere.

```suggestion
unqualifiedName = fromMaybe (error msg) . lastMay . Name.segments' . name
  where msg = "Var.unqualifiedName applied to a variable whose name is the empty string"
```

In principle there's nothing wrong with a function like this silently returning the empty string (much like `take 10` may return a list with fewer than 10 elements and that's all fine and expected) but it's a change to the conventions previously established for this function, so unless you want to audit all the use sites to make sure they are okay with the new convention I'd leave it alone.

Reply:
This function throwing an error was the source of a bug in `ls` which caused UCM to crash. We have a type system though, so we can just force callers to give us at least one segment here.

Reply:
Gotcha. I guess you could have this function return `Maybe` and update the callers, but I think your original change is fine too.